mooncake-store/src/master_service.cpp

```diff
@@ -200,6 +186,13 @@ ErrorCode MasterService::GetReplicaList(
     // Only mark for GC if enabled
     if (enable_gc_) {
         MarkForGC(key, 1000);  // After 1 second, the object will be removed
+    } else {
+        // Grant a lease to the object so it will not be removed
+        // when the client is reading it.
+        metadata.lease_timeout = std::max(
```

Comment:
Would a `grant_lease` interface help reduce the duplicated code here?

Reply:
Sounds great